testing/internal/e2e/tests/static/key_destruction_test.go

```diff
@@ -0,0 +1,344 @@
+package static_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/api/authmethods"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestKeyDestructionCli uses the boundary CLI to test key destruction.
+func TestKeyDestructionCli(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"scopes", "create",
+			"-name", "testscope",
+			"-format", "json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var scopeReply scopes.ScopeCreateResult
+	require.NoError(t, json.Unmarshal(output.Stdout, &scopeReply))
+	t.Cleanup(func() {
+		output = e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs(
+				"scopes", "delete",
+				"-id", scopeReply.Item.Id,
+				"-format", "json",
+			),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"scopes", "list-keys",
+			"-scope-id", scopeReply.Item.Id,
+			"-format", "json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var keys scopes.KeyListResult
+	err := json.Unmarshal(output.Stdout, &keys)
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, len(kms.ValidDekPurposes())+1)
```

Comment:
Also, would it be worth listing out the keys on failure to assist with debugging?

Reply:
Sure, I've added a debug message, though I don't know how useful it actually be since `keys.Items` is a pretty complex structure.